### PR TITLE
Nut00a

### PR DESCRIPTION
**Pull request by prakharcode**

### CODE REVIEW

File: src/AstroBase.jl [hunk 1]
```diff
@@ -485,4 +487,85 @@ function xy06(jd1, jd2)
...
+    alea = mean_longitude(earth, t)
+    alma = mean_longitude(mars, t)
+    alju = mean_longitude(jupiter, t)
+    alsa = mean_longitude(saturn, t)
+    alur = mean_longitude(uranus, t)
+
+    elp = sec2rad(mod((@evalpoly t 1287104.79305 129596581.0481 -0.5532 0.000136 -0.00001149), ARCSECONDS_IN_CIRCLE))
+    d   = sec2rad(mod((@evalpoly t 1072260.70369 1602961601.2090 -6.3706 0.006593 -0.00003169), ARCSECONDS_IN_CIRCLE))
+
+    al   = mod(2.35555598 + 8328.6914269554 * t, 2pi)
```
Comment (giordano) [inline src/AstroBase.jl]: Use mod2pi, also below

Comment (prakharcode) [reply src/AstroBase.jl]: `mod2pi` was giving errors when tested with `ERFA.nut00a` :worried:

Comment (giordano) [reply src/AstroBase.jl]: Please, explain what kind of errors

Comment (prakharcode) [reply src/AstroBase.jl]: oh, I tried to generate the errors again, it's fixed, I guess it was due to another expression which I fixed together with removing `mod2pi` so I was in an impression that `mod2pi `'s regularized output might be of problem.
It's solved. :smile:

File: src/AstroBase.jl [hunk 2]
```diff
@@ -485,4 +487,85 @@ function xy06(jd1, jd2)
...
+        sarg, carg = sincos(arg)
+        dp += (xls_nutation[i][6] + xls_nutation[i][7] * t) * sarg + xls_nutation[i][8] * carg
+        de += (xls_nutation[i][9] + xls_nutation[i][10] * t) * carg + xls_nutation[i][11] * sarg
+    end
+
+    dpsils, depsls = dp * U2R, de * U2R
+
+    dp, de = 0.0, 0.0
+    for i in reverse(eachindex(xpl_nutation))
+        arg = mod(xpl_nutation[i][1] * al   +
```
Comment (helgee) [inline src/AstroBase.jl]: `mod2pi`

File: src/nut_const.jl [hunk 3]
```diff
@@ -0,0 +1,1646 @@
+xls_nutation = [
```
Comment (helgee) [inline src/nut_const.jl]: This should be a tuple of tuples.

File: src/AstroBase.jl [hunk 4]
```diff
@@ -485,4 +487,85 @@ function xy06(jd1, jd2)
...
+                    xls_nutation[i][2] * elp +
+                    xls_nutation[i][3]  * f +
+                    xls_nutation[i][4]  * d +
+                    xls_nutation[i][5] * om, 2pi)
+        sarg, carg = sincos(arg)
+        dp += (xls_nutation[i][6] + xls_nutation[i][7] * t) * sarg + xls_nutation[i][8] * carg
+        de += (xls_nutation[i][9] + xls_nutation[i][10] * t) * carg + xls_nutation[i][11] * sarg
+    end
+
+    dpsils, depsls = dp * U2R, de * U2R
```
Comment (helgee) [inline src/AstroBase.jl]: I would prefer this expression: `dpsils, depsls = sec2rad(dp * 1e-7), sec2rad(de * 1e-7)`

This is more characters in total but the intention is clear from looking at the line. If you introduce the constant I need to scroll up to understand what it does.

File: src/AstroBase.jl [hunk 5]
```diff
@@ -485,4 +487,85 @@ function xy06(jd1, jd2)
...
+             xpl_nutation[i][12] * alne +
+             xpl_nutation[i][13] * apa, 2pi)
+        sarg, carg = sincos(arg)
+
+        dp += xpl_nutation[i][14] * sarg + xpl_nutation[i][15] * carg
+        de += xpl_nutation[i][16] * sarg + xpl_nutation[i][17] * carg
+
+    end
+
+    dpsipl = dp * U2R
```
Comment (helgee) [inline src/AstroBase.jl]: See my other comment.

**Review by helgee — CHANGES_REQUESTED** [review]

**Pull request merged**